src/Servers/IIS/IIS/test/testassets/InProcessWebSite/Startup.cs

```diff
@@ -1135,6 +1135,191 @@ public Task ResponseTrailers_NullValues_Ignored(HttpContext context)
             return Task.FromResult(0);
         }
 
+        public Task AppException_BeforeResponseHeaders_500(HttpContext context)
+        {
+            throw new Exception("Application exception");
+        }
+
+        public async Task AppException_AfterHeaders_PriorOSVersions_ResetCancel(HttpContext httpContext)
+        {
+            await httpContext.Response.Body.FlushAsync();
+            throw new Exception("Application exception");
+        }
+
+        public async Task AppException_AfterHeaders_ResetInternalError(HttpContext httpContext)
+        {
+            await httpContext.Response.Body.FlushAsync();
+            throw new Exception("Application exception");
+        }
```

Comment:
Can you combine some of these? The app code is the same, only the behavior observed on the client is different.

Reply:
I purposely didn't combine them to keep them 1 to 1 with the tests. IMO it makes it easier to match test to the part of the startup.